packages/api/src/task/scheduler.ts

```diff
@@ -406,15 +406,25 @@ export class TaskScheduler {
 
   async scheduleTask(task: WithID<Task>, retries = 0) {
     const timestamp = Date.now();
-    await this.updateTask(task, {
-      // only update scheduledAt on the first schedule (retries == 0)
-      scheduledAt: retries ? undefined : timestamp,
-      status: {
-        phase: "waiting",
-        updatedAt: timestamp,
-        retries,
-      },
-    });
+    try {
```

Comment:
wonder if a single try block wouldnt be simpler. then we can log the error and can figure out where it came from, and fail the task with a generic message like "Failed to enqueue task"

Reply:
exactly!